src/engine/renderer/glsl_source/reliefMapping_fp.glsl

```diff
@@ -56,12 +56,47 @@ vec3 NormalInTangentSpace(vec2 texNormal)
 	//
 	// crunch -dxn seems to produce such files, since alpha channel is abused such format
 	// is unsuitable to embed height map, then height map must be distributed as loose file
+	// -- illwieckz
 	normal = texture2D(u_NormalMap, texNormal).rga;
 	normal.x *= normal.z;
 	normal.xy = 2.0 * normal.xy - 1.0;
-	normal.z = sqrt(1.0 - dot(normal.xy, normal.xy));
+	// In a perfect world this code must be enough:
+	// normal.z = sqrt(1.0 - dot(normal.xy, normal.xy));
+
+	// Unvanquished texture known to trigger black normalmap artifacts
+	// when doing Z reconstruction:
+	//   textures/shared_pk02_src/rock01_n
+	// -- illwieckz
+
+	// I checked the DXT5 normal maps on Chasm for out-of-range values.
```

Comment:
All of the following discussion is a bit verbose; how about this:

Although the normal vector is supposed to have a length of 1, dot(normal.xy, normal.xy) may be greater than 1 due to compression artifacts: values as large as 1.27 have been observed with crunch --<insert options here>. This might happen with other formats too. So we must take care not to take the square root of a negative number here.